app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -38,76 +70,175 @@ class FireDatabase : Database() {
+        val future = CompletableFuture<UserModel>()
+
+        database.child(usersProfileFileName).child(userId).get().addOnSuccessListener { userData ->
+            if(userData == null) {
+                future.completeExceptionally(java.lang.Error("There is no user account corresponding to this userId."))
+            }else{
+                if(userAuth==null){
+                    future.completeExceptionally(java.lang.Error("There is no user logged on the app."))
+                }else{
+                    val firstname = userData.child("firstname").value
+                    val surname = userData.child("surname").value
+                    val dateOfBirth = userData.child("dateOfBirth").value
+                    val distanceGoal = userData.child("distanceGoal").value
+                    val activityTimeGoal = userData.child("activityTimeGoal").value
+                    val nbOfPathsGoal = userData.child("nbOfPathsGoal").value
+                    if(firstname==null||surname==null||dateOfBirth==null||distanceGoal==null||activityTimeGoal==null||nbOfPathsGoal==null){
+                        future.completeExceptionally(java.lang.Error("The user account present on the database is incomplete."))
+                    }else{
+                        future.complete(
+                            UserModel(userAuth, firstname as String, surname as String, LocalDate.ofEpochDay(dateOfBirth as Long),
+                                    distanceGoal as Double, activityTimeGoal as Double, nbOfPathsGoal as Int, FireDatabase()))
+                    }
+                }
+            }
+        }.addOnFailureListener{
+            future.completeExceptionally(it)
+        }
+        return future
     }
 
-    override fun setUserGoals(
-        username: String,
-        distanceGoal: Double,
-        timeGoal: Double,
-        nbOfPathsGoal: Int
-    ) {
-        val userAdd = HashMap<String, String>()
-        userAdd.put("distanceGoal", distanceGoal.toString())
-        userAdd.put("activityTimeGoal", timeGoal.toString())
-        userAdd.put("numberOfPathsGoal", nbOfPathsGoal.toString())
-        updateUserData(userAdd, username)
+    override fun getLoggedUserAccount(): CompletableFuture<UserModel> {
+        val userId = getUserId()
+        return getUserAccount(userId)
     }
 
-    override fun setDistanceGoal(userId: String, distanceGoal: Double): CompletableFuture<Boolean> {
-        TODO("Will be done in the next task: database rebase")
+    override fun setDistanceGoal(distanceGoal: Double): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val dataUpdated = HashMap<String, Any>()
+        dataUpdated.put("distanceGoal", distanceGoal)
+
+        if(distanceGoal <= 0.0){
+            throw java.lang.Error("The distance goal can't be less or equal than 0.")
+        }
+
+        database.child(usersProfileFileName).child(getUserId()).updateChildren(dataUpdated)
```

Comment:
`database.child(usersProfileFileName).child(getUserId()).updateChildren(dataUpdated)` is copied 3 times in this file. It should be refactored to a method `updateUser(uid: String, data: Map<String, Any>)`.